src/utils/config/authentication/providers/authentication.provider.config.entity.ts

```diff
@@ -0,0 +1,55 @@
+import { ObjectType, Field, createUnionType } from '@nestjs/graphql';
+import { AadConfig } from './aad/aad.config.entity';
+import { IAadConfig } from './aad/aad.config.interface';
+import { IAuthenticationProviderConfig } from './authentication.provider.config.interface';
+import { SimpleAuthProviderConfig } from './simple-auth/simple-auth.provider.config.entity';
+import { ISimpleAuthProviderConfig } from './simple-auth/simple-auth.provider.config.interface';
+
+@ObjectType()
+export class AuthenticationProviderConfig
+  implements IAuthenticationProviderConfig {
+  @Field(() => String, {
+    nullable: false,
+    description: 'Name of the authentication provider.',
+  })
+  name?: string;
+
+  @Field(() => String, {
+    nullable: false,
+    description: 'Label of the authentication provider.',
+  })
+  label?: string;
+
+  @Field(() => String, {
+    nullable: false,
+    description:
+      'CDN location of an icon of the authentication provider login button.',
+  })
+  icon?: string;
+
+  @Field(() => Boolean, {
+    nullable: false,
+    description: 'Is the authentication provider enabled?',
+  })
+  enabled?: boolean;
+
+  @Field(() => AuthenticationConfigUnion, {
+    nullable: false,
+    description: 'Configuration of the authenticaiton provider',
+  })
+  config?: IAadConfig | ISimpleAuthProviderConfig;
+}
+
+export const AuthenticationConfigUnion = createUnionType({
+  name: 'ConfigUnion',
+  types: () => [AadConfig, SimpleAuthProviderConfig],
```

Comment:
This should be "AadAuthProviderConfig" at least at the graphql api level so that that is consistent. Internal names etc we can / should pick up later.

Reply:
Fixed in 222a2c77eda25806d4848fc31e19b3d6846c8aee